t/plugin/openid-connect.t

```diff
@@ -1665,3 +1665,319 @@ GET /t
+
+                    if not res then
+                        -- No response, must be an error.
+                        ngx.status = 500
+                        ngx.say(err)
+                        return
+                    elseif res.status ~= 302 then
+                        -- Not a redirect which we expect.
+                        -- Use 500 to indicate error.
+                        ngx.status = 500
+                        ngx.say("Login form submission did not return redirect to redirect URI.")
+                        return
+                    end
+
+                    -- Extract the redirect URI from the response header.
+                    -- TODO: Consider validating this against the plugin configuration.
+                    local redirect_uri = res.headers['Location']
+
+                    -- Invoke the redirect URI (which contains the authorization code as an URL parameter).
+                    res, err = httpc:request_uri(redirect_uri, {
+                            method = "GET",
+                            headers = {
+                                ["Cookie"] = cookie_str
+                            }
+                        })
+
+                    if not res then
+                        -- No response, must be an error.
+                        ngx.status = 500
+                        ngx.say(err)
+                        return
+                    elseif res.status ~= 302 then
+                        -- Not a redirect which we expect.
+                        -- Use 500 to indicate error.
+                        ngx.status = 500
+                        ngx.say("Invoking redirect URI with authorization code did not return redirect to original URI.")
+                        return
+                    end
+
+                    -- Get all cookies returned. This should update the session cookie maintained by the OIDC module with the new state.
+                    -- E.g. the session cookie should now contain the access token, ID token and user info.
+                    -- The cookie itself should however be treated as opaque.
+                    cookies = res.headers['Set-Cookie']
+
+                    -- Concatenate cookies into one string as expected when sent in request header.
+                    if type(cookies) == 'string' then
+                        cookie_str = cookies:match('([^;]*); .*')
+                    else
+                        -- Must be a table.
+                        local len = #cookies
+                        if len > 0 then
+                            cookie_str = cookies[1]:match('([^;]*); .*')
+                            for i = 2, len do
+                                cookie_str = cookie_str .. "; " .. cookies[i]:match('([^;]*); .*')
+                            end
+                        end
+                    end
+
+                    -- Get the final URI out of the Location response header. This should be the original URI that was requested.
```

Comment:
The similar Lua snipper is repeated in multiple tests. Could we refactor it?

Reply:
https://github.com/apache/apisix/blob/acbb111afbbd096444bdd6dc0307d18187b19829/t/plugin/skywalking.t#L30
You can define helper functions in extra_init_by_lua

Reply:
Fixed. Please review it again.

Reply:
Fixed. Please review it again.